### ISSUE TITLE
Migrate to Jest

### ISSUE THREAD

**Issue by Illyism** [marktext/marktext#1315]
### Question description

Is there any particular reason this project is not using Jest?

- It would simplify unit tests for vue
- Coverage, config is simpler
- I would move the tests to be closer to the file; eg in a __tests__ folder in the same directory
- There seem to be solutions so far for running electron in jest

**Comment by Jocs** [comment]
>Is there any particular reason this project is not using Jest?

No, I also prefer use Jest, but this priority is not very high relative to other issues, so we have not had time to optimize.

PR is welcomed if you can help with this.

**Comment by Illyism** [comment]
I'll take a look in the next few days

**Comment by Jocs** [comment]
Thanks for the suggestion back in 2019! The project has since adopted Vitest (rather than Jest) as its unit test framework — it covers all the goals mentioned here: simpler Vue integration, built-in coverage, and a cleaner config. Vitest pairs naturally with our electron-vite build system. Closing this as the underlying need has been fulfilled.
